services/QuillLMS/app/workers/sync_vitally_past_year_accounts_worker.rb

```diff
@@ -0,0 +1,10 @@
+class SyncVitallyPastYearAccountsWorker
+  include Sidekiq::Worker
+
+  def perform(school_ids, year)
+    schools = School.where(id: school_ids)
+    schools.each do |school|
+      PreviousYearSchoolDatum.new(school, year).calculate_and_save_data
```

Comment:
I would create a new worker here:
`CachePreviousYearSchoolDatumWorker.perform_async`.  This keeps the individual jobs very small but also if one of them fails it doesn't restart the whole `SyncVitallyPastYearAccountsWorker`